src/test/java/org/jenkinsci/plugins/casc/DataBoundConfiguratorTest.java

```diff
@@ -24,11 +26,11 @@
 
     @Test
     public void configure_databound() throws Exception {
-        Map<String, Object> config = new HashMap<>();
-        config.put("foo", "foo");
-        config.put("bar", true);
-        config.put("qix", 123);
-        config.put("zot", "DataBoundSetter");
+        Mapping config = new Mapping();
+        config.put("foo", new Scalar("foo"));
```

Comment:
Why not modify the put<String, String> to convert string to Scalar?

Reply:
as Mapping is a Map<String, CNode> java will fail with a cast exception if invoked with anything but a CNode. Also we only use such an explicit `put()` in few test-cases, we prefer to load a sample yaml file for integration tests.

Reply:
.. but we can offer overridden method

Reply:
Awesome looks better 👍